examples/table.rs

```diff
@@ -161,6 +236,82 @@ fn ui(f: &mut Frame, app: &mut App) {
         bar.into(),
         "".into(),
     ]))
+    .bg(app.colors.buffer_bg)
     .highlight_spacing(HighlightSpacing::Always);
-    f.render_stateful_widget(t, rects[0], &mut app.state);
+    f.render_stateful_widget(t, area, &mut app.state);
+}
+
+fn constraint_len_calculator(items: &[Vec<String>]) -> (u16, u16, u16) {
+    let len_vec = items
+        .iter()
+        .map(|row| {
+            (
+                row[0].chars().count() as u16,
+                row[1]
+                    .split('\n')
+                    .map(|address_line| address_line.chars().count())
+                    .max()
+                    .unwrap() as u16,
+                row[2].chars().count() as u16,
+            )
+        })
+        .collect::<Vec<(u16, u16, u16)>>();
+
+    let longest_name_len = len_vec.iter().map(|row| row.0).max().unwrap();
```

Comment:
If you're iterating and collecting then iterating on the result 3 times or might be neater to iterate and collect the max for each column in a single iteration.